common/lib/dependabot/pull_request_creator/message_builder/link_and_mention_sanitizer.rb

```diff
@@ -54,9 +58,23 @@ def sanitize_links_and_mentions(text:, unsafe: false, format_html: true)
           sanitize_nwo_text(doc)
 
           mode = unsafe ? :UNSAFE : :DEFAULT
-          return doc.to_commonmark([mode] + COMMONMARKER_OPTIONS) unless format_html
 
-          doc.to_html(([mode] + COMMONMARKER_OPTIONS), COMMONMARKER_EXTENSIONS)
+          # If the text does not contain markdown, we need to use the HARDBREAKS option
+          commonmarker_render_options = if text.length > 1000
+                                          [COMMONMARKER_OPTIONS]
+                                        else
+                                          # If text length is manageable, use regex to determine rendering options
+                                          (unless text.match?(MARKDOWN_REGEX)
+                                             COMMONMARKER_OPTIONS + [:HARDBREAKS]
+                                           end) || [COMMONMARKER_OPTIONS]
+                                        end
+
+          options = [mode] + commonmarker_render_options
+          options = options.flatten.uniq
```

Comment:
Curious, why'd you have to do this? Since it wasn't there previously, just surprised to see it... Was the duplicate present previously and we should have been doing this all along? Or does this new code somehow result in duplicates that need to be cleaned up?

Reply:
This code was related to the regex timeout logic. Since the timeout is not needed, I've discarded lines 72 and 73.